extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js

```diff
@@ -0,0 +1,132 @@
+define('wikia.articleVideo.featuredVideo.events', function () {
+	var state = getNewState(),
+		wasAlreadyUnmuted = false,
+		depth = 0,
+		prefixes = {
+			ad: 'ad',
+			video: 'video'
+		},
+		isPlayerPaused = false,
+		playerInstance;
+
+	function getDefaultState() {
+		return {
+			wasFirstQuartileTriggered: false,
+			wasMidPointTriggered: false,
+			wasThirdQuartileTriggered: false,
+			progress: {
+				durationWatched: 0,
+				percentWatched: 0
+			}
+		}
+	}
+
+	function getNewState() {
+		return {
+			ad: getDefaultState(),
+			video: getDefaultState()
+		}
+	}
+
+	function handleTime(prefix, data) {
+		var positionFloor = Math.floor(data.position),
+			percentPlayed = Math.floor(positionFloor * 100 / data.duration);
+
+		if (positionFloor > state[prefix].progress.durationWatched && positionFloor % 5 === 0) {
+			playerInstance.trigger(prefix + 'SecondsPlayed', { value: positionFloor });
+
+			state[prefix].progress.durationWatched = positionFloor;
+		}
+
+		if (percentPlayed >= 25 && !state[prefix].wasFirstQuartileTriggered) {
+			playerInstance.trigger(prefix + 'FirstQuartile');
+			state[prefix].wasFirstQuartileTriggered = true;
+		}
+
+		if (percentPlayed >= 50 && !state[prefix].wasMidPointTriggered) {
+			playerInstance.trigger(prefix + 'MidPoint');
+			state[prefix].wasMidPointTriggered = true;
+		}
+
+		if (percentPlayed >= 75 && !state[prefix].wasThirdQuartileTriggered) {
+			playerInstance.trigger(prefix + 'ThirdQuartile');
+			state[prefix].wasThirdQuartileTriggered = true;
+		}
+
+		if (percentPlayed > state[prefix].progress.percentWatched && percentPlayed % 10 === 0) {
+			playerInstance.trigger(prefix + 'PercentPlayed', { value: percentPlayed });
+
+			state[prefix].progress.percentWatched = percentPlayed;
+		}
+	}
+
+	return function (providedPlayerInstance, willAutoplay) {
+		playerInstance = providedPlayerInstance;
+
+		console.info('jwplayer before ready');
+		playerInstance.once('ready', function () {
+			console.info('jwplayer player ready');
+			var relatedPlugin = playerInstance.getPlugin('related');
+
+			relatedPlugin.on('open', function () {
+				console.info('jwplayer related plugin open');
+				playerInstance.trigger('relatedVideoImpression');
+				state[prefixes.video] = getDefaultState();
+			});
+
+			relatedPlugin.on('play', function (data) {
+				console.info('jwplayer related plugin play');
+				depth++;
+
+				playerInstance.trigger('relatedVideoPlay', {
+					auto: data.auto,
+					item: data.item,
+					position: data.position,
+					depth: depth
+				});
+			});
+		});
+
+		playerInstance.on('play', function (data) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "data". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)